src/domain/headless-cms/__tests__/utils.test.ts

```diff
@@ -390,3 +414,298 @@ describe('rewriteInternalURLs', () => {
+    {
+      locale: 'fi',
+      currentPath: '/cms-page/test-page',
+      itemPath: '/test-page/',
+      expected: true,
+      description: 'FI: Exact match',
+    },
+    {
+      locale: 'fi',
+      currentPath: '/cms-page/test-page/sub-page',
+      itemPath: '/test-page/',
+      expected: true,
+      description: 'FI: Current path is sub-path',
+    },
+    {
+      locale: 'fi',
+      currentPath: '/cms-page/another-page',
+      itemPath: '/test-page/',
+      expected: false,
+      description: 'FI: Different page',
+    },
+    {
+      locale: 'fi',
+      currentPath: '/cms-page/test-page',
+      itemPath: '/fi/test-page/',
+      expected: true,
+      description: 'FI: Item path has locale prefix (should be stripped)',
+    },
+    // English
+    {
+      locale: 'en',
+      currentPath: '/en/cms-page/test-page',
+      itemPath: '/test-page/',
+      expected: true,
+      description: 'EN: Exact match',
+    },
+    {
+      locale: 'en',
+      currentPath: '/en/cms-page/test-page/sub-page',
+      itemPath: '/test-page/',
+      expected: true,
+      description: 'EN: Current path is sub-path',
+    },
+    {
+      locale: 'en',
+      currentPath: '/en/cms-page/another-page',
+      itemPath: '/test-page/',
+      expected: false,
+      description: 'EN: Different page',
+    },
+    {
+      locale: 'en',
+      currentPath: '/en/cms-page/test-page',
+      itemPath: '/en/test-page/',
+      expected: true,
+      description: 'EN: Item path has locale prefix (should be stripped)',
+    },
+    // Swedish
+    {
+      locale: 'sv',
+      currentPath: '/sv/cms-page/test-page',
+      itemPath: '/test-page/',
+      expected: true,
+      description: 'SV: Exact match',
+    },
+    // Edge cases
+    {
+      locale: 'fi',
+      currentPath: '/cms-page/test-page',
+      itemPath: null,
+      expected: false,
+      description: 'Null item path',
+    },
+    {
+      locale: 'fi',
+      currentPath: '/cms-page/test-page',
+      itemPath: undefined,
+      expected: false,
+      description: 'Undefined item path',
+    },
+    {
+      locale: 'fi',
+      currentPath: '/cms-page/root',
+      itemPath: '/',
+      expected: true,
+      description: 'Root path item',
+    },
+  ])(
+    'getIsItemActive returns $expected for $description',
+    ({ locale, currentPath, itemPath, expected }) => {
+      mockLocation(currentPath);
+      // Use the mock getCmsPagePath or ensure the real one is available
+      const item = { ...menuItem, path: itemPath };
+
+      // Assertion on the function's behavior
+      expect(getIsItemActive(item, locale)).toBe(expected);
+    }
+  );
+
+  it('should handle window being undefined (SSR)', () => {
+    const originalWindow = global.window;
+    // @ts-expect-error: Simulate SSR environment where window is undefined
+    delete global.window;
+    expect(getIsItemActive(menuItem, 'fi')).toBe(false);
+    global.window = originalWindow; // Restore window
```

Comment:
Might the temporary deletion of global.window interact badly with other tests being run if they were run just in the right moment? Would mocking it be safer here?